### PR TITLE
feat(core): secure customer s3 bucket access for backups using cross-account role

### PR DESCRIPTION
**Pull request by chandumlg**
# Description

If BACKUP_IAM_ROLE_ARN env is specified, sts creds are generated assuming the role for backups to customer bucket. BACKUP_IAM_ROLE_ARN is the ARN of the role the customer creates with a trust policy principal being our AWS acccount and allowing necessary S3 actions on their bucket

More details: https://rudderlabs.slack.com/archives/C03KSBL7JM7/p1657156293344469

## Notion Ticket

https://www.notion.so/rudderstacks/Secure-connection-to-customer-S3-buckets-for-backups-47fa67c7cd1c40fc9edbdf9118a39703

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by psrikanth88 — APPROVED** [review]

**Comment by koladilip** [comment]
@chandumlg I have resolved the conflicts, please check once from your end.

**Comment by koladilip** [comment]
@chandumlg I have test my changes in this branch on dev cluster, working as expected with IAM role.

File: services/filemanager/s3manager.go [hunk 1]
```diff
@@ -192,6 +192,7 @@ func (manager *S3Manager) getSession(ctx context.Context) (*session.Session, err
 		}
 		manager.Config.Region = aws.String(region)
 	}
+
 	if manager.Config.Region == nil {
```
Comment (chandumlg) [inline services/filemanager/s3manager.go:196]: Is this block needed? Will `manager.Config.Region` ever be nil?

Comment (chandumlg) [inline services/filemanager/s3manager.go:196]: Is this block needed? Will manager.Config.Region ever be nil?

Comment (koladilip) [reply services/filemanager/s3manager.go:196]: Make sense, it is not needed, we can remove it.

Comment (koladilip) [reply services/filemanager/s3manager.go]: Removed it

**Review by koladilip — APPROVED** [review]

**Pull request merged**